pynsights/view.py

```diff
@@ -20,23 +19,27 @@
 lastCall = {}
 when = 0
 
-dump_filename = None
-index_output_filename = None
-index_input_filename = __file__.replace("view.py", "index.html")
+template_file = __file__.replace("view.py", "index.html")
 
 
-def read_dump():
+def show_progress(percent, end=""):
+    print(f"\rPynsights: rendering, {percent}% done", end=end)
+
+
+def read_dump(input_file):
     done = 0
-    with open(dump_filename) as fp:
+    with open(input_file) as fp:
         lines = fp.readlines()
         one_percent = round(len(lines) / 100)
         for n, line in enumerate(lines):
             handle_line(line)
             if not one_percent or n % one_percent == 0:
                 if done % 10 == 0:
-                    print("%d%% done" % done)
+                    show_progress(done)
                 done += 1
         flushCallSites()
+    show_progress(100, "\n")
```

Comment:
I noticed that in some cases we would not reach 100 (rounded division).
This makes sure we always show a progress of 100% at the end.